### PR TITLE
Add search feature with typesense integeration

### PR DESCRIPTION
**Pull request by kimz95**
### Changes Made:
- Implemented search functionality using Typesense.
- Added API endpoint for searching jobs.
- Added relevant tests.
- Removed unnecessary prints.

### Environment Variables
Please ensure the following values are added to GitHub secrets before running:

- `TYPESENSE_URL`: The base URL for the Typesense server (e.g., `http://localhost:8108`)
- `TYPESENSE_API_KEY`: A secret key used for authentication

### CODE REVIEW

**Comment by dIB59** [comment]
@kimz95 you have to resolve the merge request before asking for a review

**Comment by kimz95** [comment]
> @kimz95 you have to resolve the merge request before asking for a review

Apologies. The merge conflicts have been resolved

File: webserver/src/macros_local.rs [hunk 1]
```diff
@@ -36,3 +36,19 @@ macro_rules! run_async_query {
...
+        use reqwest::Error;
+        use actix_web::error::ErrorInternalServerError;
+
+        web::block(move || {
+            let response = $query(&$search_state,  $url, $user).map_err(Error::from);
+            response
+        })
+        .await
+        .map_err(ErrorInternalServerError)
+        .expect("internal server error")
```
Comment (Taza53) [inline webserver/src/macros_local.rs:51]: Don't use ".expect()" use "?"

And we probably don't need this macro, as run_async_query does the same thing

File: webserver/src/handlers/job_handler.rs [hunk 2]
```diff
@@ -120,7 +115,8 @@ pub async fn create_job(
     user_sub: UserSub,
     pool: web::Data<DbPool>,
     job_req: web::Json<CreateJobRequest>,
-) -> Result<impl Responder, impl ResponseError> {
```
Comment (Taza53) [inline webserver/src/handlers/job_handler.rs:123]: This change break the application, revert it.

**Review by Taza53 — CHANGES_REQUESTED** [review]

**Comment by Taza53** [comment]
Please run `cargo test` and make sure all test pass and code compiles.

**Comment by dIB59** [comment]
Please mark this as a draft if the pull request is not ready.

File: webserver/src/macros_local.rs (hunk 1 above)
Comment (kimz95) [reply webserver/src/macros_local.rs]: > Don't use ".expect()" use "?"

This has been addressed in the new pull request.
I believe we need the new marco for handling different return error types.

File: webserver/src/handlers/job_handler.rs (hunk 2 above)
Comment (kimz95) [reply webserver/src/handlers/job_handler.rs]: This is addressed in the new pull request.

**Comment by kimz95** [comment]
> Please run `cargo test` and make sure all test pass and code compiles.

Individual runs of 'cargo test' used to run successfully. However, after multiple runs of tests, there were occasional failures because of two issues addressed in the new PR. If the test cases don't run at all, please ensure that "docker compose" is re-run and the environment variables are added.

**Pull request closed without merging**